### PR TITLE
Nested-channels related GUI improvements

### PR DESCRIPTION
**Pull request by ichorid**
Adds breadcrumbs for navigation and fixes some bugs.

### CODE REVIEW

**Comment by ichorid** [comment]
retest this please

**Comment by xoriole** [comment]
retest this please

**Comment by xoriole** [comment]
retest this please

**Comment by xoriole** [comment]
retest this please

**Comment by ichorid** [comment]
retest this please

**Comment by xoriole** [comment]
For the PRs involving an update to Qt/GUI, let's add to the PR checklist a screenshot of the GUI.

File: src/tribler-gui/tribler_gui/widgets/channelcontentswidget.py [hunk 1]
```diff
@@ -290,7 +294,19 @@ def update_labels(self, dirty=False):
...
         self.category_selector.setHidden(root and (discovered or personal_model))
         # initialize the channel page
-        self.channel_name_label.setText(self.model.channel_info['name'])
+
+        # Assemble the channels navigation breadcrumb by utilising RichText links feature
+        self.channel_name_label.setTextFormat(Qt.RichText)
+        breadcrumb_text = ""
+        for m, model in enumerate(self.channels_stack[:-1]):
+            breadcrumb_text += (
+                f'<a style="text-decoration:none;color:#A5A5A5;" href="{m}">{model.channel_info["name"]}</a>'
```
Comment (xoriole) [inline src/tribler-gui/tribler_gui/widgets/channelcontentswidget.py]: Anchor `href` navigation works in Qt?

Comment (ichorid) [reply src/tribler-gui/tribler_gui/widgets/channelcontentswidget.py]: It is possible to add RichText formatting to RichText QT text type objects. That includes hyperlinks. And one can catch hyperlink signals from clicking the links.

**Comment by ichorid** [comment]
>Could we shorten the breadcrumb by adding an ellipsis to the left of the text?
Good point. I'll try to do it now.

> adding a back button would still be very valuable and help navigation (it takes less mental effort to click the back button than search for the right level in the breadcrumb you want to go to).
Again, good point. I'll bring the back button back.

> 
> I also noticed that folders are being marked as torrent, see https://gyazo.com/373b7466360066927f4c082bc571bbda. This leads to an incorrect indication of how many torrents there are in ones channel.

Yep, that's a bug. I'll try to address it in this PR.

**Comment by ichorid** [comment]
Well, channels/folders being marked as torrents is not a bug. It is correctly showing the total number of objects in the current view. We need to discuss this thing from the UI standpoint. Other than that, I consider this PR good to go.

**Pull request merged**